include/veriblock/blockchain/btc_block_addon.hpp

```diff
@@ -13,32 +13,81 @@
 namespace altintegration {
 
 struct BtcBlockAddon {
+  using ref_height_t = int32_t;
+
   //! (memory only) total amount of work in the chain up to and including this
   //! block
   ArithUint256 chainWork = 0;
 
   //! reference counter for fork resolution
-  uint32_t refCounter = 0;
+  std::vector<ref_height_t> refs{};
```

Comment:
Maybe use a map instead?
std::unordered_map<ref_height_t, int32_t> refs{}

Reply:
keys may be duplicated

Reply:
Increase ref counter for duplicate keys

Reply:
not sure it's worth it. in practice VTB contexts aren't going to overlap much